### PR TITLE
9_ building_data-model and 8_room_data_model

### PR DESCRIPTION
**Pull request by Max784**

### CODE REVIEW

File: app/models/building.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Building < ApplicationRecord
+    include Timeable
+end
```
Comment (bewee) [inline app/models/building.rb]: Looks a bit empty, aren't some fields missing here? 🤔

File: db/schema.rb [hunk 2]
```diff
@@ -27,4 +57,5 @@
     t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
   end
 
+  add_foreign_key "rooms", "buildings"
```
Comment (bewee) [inline db/schema.rb:60]: Why is there no foreign key between users and buildings?

File: spec/factories/openingtimes.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+  factory :openingtime do
+    opens { "2021-11-29 07:41:48" }
+    closes { "2021-11-29 07:41:48" }
```
Comment (bewee) [inline spec/factories/openingtimes.rb:4]: Why does it store the date, not only the time? 
Also, I would prefer the sample being opened for at least one second ;) 
(Also, maybe add validation that closes - opens is > 0)

File: spec/factories/rooms.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+  factory :room do
+    association :building
+    name { "C.2.4" }
+    floor { "2" }
+    room_type { "Bachelorproject office" }
+    contact_person { "Jonas Cremerius" }
```
Comment (bewee) [inline spec/factories/rooms.rb:7]: Should be reference to (already existing) users relation

File: spec/helpers/buildings_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+#
+# describe BuildingsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe BuildingsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (bewee) [inline spec/helpers/buildings_helper_spec.rb:14]: Pending tests, but we could still fill these in a later PR

File: spec/models/openingtime_spec.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Openingtime, type: :model do
+  it "does not fail when created with factory bot" do
+    openingtime = FactoryBot.create(:openingtime)
+    expect(openingtime.day).to eq(1)
```
Comment (bewee) [inline spec/models/openingtime_spec.rb:6]: Should also test the other fields

File: spec/models/room_spec.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    building = FactoryBot.create :building
+
+    expect(room.name).to eq('C.2.4')
+    expect(room.floor).to eq("2")
+    expect(room.building.name).to eq(building.name)
+    expect(room.room_type).to eq('Bachelorproject office')
+    expect(room.contact_person).to eq('Jonas Cremerius')
+  end
+
+  it "has a building" do
```
Comment (bewee) [inline spec/models/room_spec.rb:15]: after making user a foreign key, also add a "has a user" test

File: app/models/building.rb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+class Building < ApplicationRecord
+  include Timeable
+end
```
Comment (bewee) [inline app/models/building.rb:3]: Why is this empty? How does it work with no fields specified? o.O We are confused :D
(same for timeable.rb and rooms.rb)

**Review by bewee — CHANGES_REQUESTED** [review]
Looks good overall. We really like the solution with Timeable! Just some small things & comments

**Review by taminob — APPROVED** [review]
Validation of attributes in models and especially tests are still open and have to be implemented at a later point.
To have at least some progress, we can merge this

**Review by bewee — APPROVED** [review]

File: spec/factories/openingtimes.rb (hunk 3 above)
Comment (bewee) [reply spec/factories/openingtimes.rb]: Update: Only validation still missing

**Pull request merged**